docs/data/osdf/install-origin-container.md

```diff
@@ -0,0 +1,353 @@
+These steps will use the Pelican client
+(which you can download from <https://github.com/PelicanPlatform/pelican/releases>).
+
+If the actions fail, see debugging information in your container logs,
+which are accessible via
+```
+docker logs osdf-origin
+```
+To increase the debugging information in the origin, edit your origin configuration file and set:
+```
+Debug: true
+```
+See [this page](../../common/help.md) for requesting assistance; please include the logs in your request.
+
+1.  Optional, if DirectReads are enabled:  Download a test file (POSIX) or object (S3) directly from your origin,
+    (replacing `<TEST_PATH>` with the OSDF path to the test file or object):
+
+        :::console
+        user@host $ pelican object get 'osdf:///<TEST_PATH>?directread=1' -o /tmp/testfile
+
+    Verify the contents of `/tmp/testfile` match the test file or object your origin was serving.
+
+1.  Download a test file (POSIX) or object (S3) from your origin via a cache,
+    (replacing `<TEST_PATH>` with the OSDF path to the test file or object):
+
+        :::console
+        user@host $ pelican object get 'osdf:///<TEST_PATH>' -o /tmp/testfile2
+
+    Verify the contents of `/tmp/testfile2` match the test file or object your origin was serving.
+
+1.  Verify that your test is running against your Pelican origin:
+
+        :::console
+        user@host $ docker logs osdf-origin | grep <TEST_PATH>
+
+    Replacing `<TEST PATH>` with the same path that you used in step (1) or (2).
+    If you see output, then the OSDF is directing client requests to your Pelican origin!
+
+
+Advanced topics
+---------------
+
+### Persisting logs
+
+By default, the origin sends logs to the console, where it is captured by the container runtime's logging mechanism.
+To save logs to a file instead, set the log location in your origin config file (e.g., `/etc/osdf-origin/config.yaml`)
+as follows:
+
+```file
+Logging:
+  LogLocation: <LOG_FILE_PATH>
+```
+
+!!! danger
+    Note: the osdf-origin image does not come with log rotation; if you save logs to a file, you must
+    set up log rotation yourself to avoid running the origin out of disk space.
+
+
+### Managing an origin with systemd
+
+If you do not have a container orchestration service but still want to manage a container-based origin,
+you may run the container via a systemd service.
+The following example uses the values from the setup documented in the [Running an Origin](#running-an-origin) section above.
+
+Create the systemd service file `/etc/systemd/system/docker-osdf-origin.service` as follows:
+
+```file
+[Unit]
+Description=OSDF Origin Container
+After=docker.service
+Requires=docker.service
+
+[Service]
+TimeoutStartSec=0
+Restart=always
+ExecStartPre=-/usr/bin/docker stop osdf-origin
+ExecStartPre=-/usr/bin/docker rm osdf-origin
+ExecStartPre=/usr/bin/docker pull hub.opensciencegrid.org/pelican_platform/osdf-origin:latest
+ExecStart=/usr/bin/docker run --rm --name osdf-origin \
```

Comment:
Maybe we should suggest `--attach stdout --attach stderr` or something? That way logrotation is handled by the journal automatically

Reply:
I don't know. Pelican services (especially the ones that run xrootd) can be noisy; mixing that into their journal will drown out the other important things in there (like kernel messages). It might be worth including a link to the [Docker logging drivers configuration](https://docs.docker.com/engine/logging/configure/) page though.

Reply:
@matyasselmeci 🤷🏽  `journalctl` has a lot of niceties to filter log messages like `journalctl -k` or `journalctl --dmesg` for kernel messages

Reply:
Well yes but then we'd have to explain how to use those, instead. Also dumping xrootd logs into the journal would cause the other logs to get rotated more frequently, too.